TRAMS-API/Mapping/ODataModelHelper.cs

```diff
@@ -0,0 +1,101 @@
+﻿using API.Models.D365;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace API.Mapping
+{
+    public class ODataModelHelper<T> : IODataModelHelper<T> where T : BaseD365Model
+    {
+        private readonly Type _type;
+
+        public ODataModelHelper()
+        {
+            _type = typeof(T);
+        }
+
+        /// <summary>
+        /// Generates a list of all JsonProperty annotations set against a type's properties. Will not include @metadata extensions
+        /// Will exclude any nested classes and their annotations
+        /// </summary>
+        /// <returns>A list of the JsonProperty annotation - excludes any @metadata extensions</returns>
+        public List<string> GetSelectClause()
+        {
+            var jsonProps = _type.GetProperties()
+                                 .Where(p => p.GetCustomAttribute<JsonPropertyAttribute>() != null && IsSystemType(p))
+                                 .Select(p => p.GetCustomAttribute<JsonPropertyAttribute>().PropertyName.Split("@").First())
+                                 .ToList();
+
+            var props = _type.GetProperties().ToList();
+
+            foreach (var prop in props)
+            {
+                var type = prop.GetType();
+                var isValueType = type.IsValueType;
+            }
+
+            return jsonProps;
+        }
+
+        private static bool IsSystemType(PropertyInfo p)
+        {
+            return p.PropertyType.FullName.Contains("System.");
+        }
+
+        public string GetExpandClause()
+        {
+            var properties = _type.GetProperties()
+                                   .Where(p => p.GetCustomAttribute<JsonPropertyAttribute>() != null && 
+                                              !IsSystemType(p))
+                                   .ToList();
+
+            var individualExpandClauses = new List<string>();
+
+            foreach(var property in properties)
+            {
+                var jsonProps = property.PropertyType.GetProperties()
```

Comment:
Suggestion: I've seen the same method of getting properties coded in two different ways in this class.  Might be worth breaking out into a small sub-routine.